### PR TITLE
Filetransfer

### PR DESCRIPTION
**Pull request by perassijulian**
Should be fully working. Have in mind that to send a file to another address they need to be verified in XMTP. This is done when you enter the filetransfer tab and sign the 2 metamask requests.
You can send files to yourself and see them in real time since streaming is working.
Also I have disabled reactStrictMode so it doesn't re-render the use effect.

### CODE REVIEW

**Comment by 3lLobo** [comment]
Wouldn't it be better if we verify the user to XMTP right away on login?

File: app/contactSlice.js [hunk 1]
```diff
@@ -56,6 +56,18 @@ export const contactSlice = createSlice({
...
+        } else {
+          const newProfile = emptyProfile(
+            action.payload.address,
+            action.payload.isOneHop
+          )
+          console.log(
+            '🚀 ~ file: contactSlice.js ~ line 61 ~ newProfile',
+            newProfile
+          )
+          state.contactInEdit = newProfile
```
Comment (3lLobo) [inline app/contactSlice.js]: Nonono, we not changing this anymore 😆
I fixed it yesterday `contacts` is not gonna be `null` anymore.

File: components/FileTransfer/index.js [hunk 2]
```diff
@@ -231,13 +241,14 @@ const FileTransfer = () => {
                 <Avatar scale={110} />
               </div>
               <div className="mr-11  col-span-2 div-black dark:div-indigo-50 self-center ">
-                {' '}
-                {m.sender}{' '}
+                {/**WTF COVALENT IS BRINGING THE ADDRESS LOWERCASES!! */}
+                <AddressTag address={m.sender.toLowerCase()} isOneHop={true} />
```
Comment (3lLobo) [inline components/FileTransfer/index.js:245]: The `isOneHop` is a flag for contacts which the user had a transaction with before, based on that flag the user can give a public tag to that address or not.
Thus, only use it when  the contact is added from the `transactions` tab.

File: components/SideBar/index.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {
 import ProfileCard from '../Profile/profileCard'
 import { Transactions } from '../Transactions'
 import { useState } from 'react'
-import { ContactSlice } from '../Profile/contactSlice'
+import { ContactSlices } from '../contactSlices'
 import FileTransfer from '../FileTransfer'
```
Comment (3lLobo) [inline components/SideBar/index.js]: This is already fixed. Merge main into your branch first ;)

File: next.config.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
-  reactStrictMode: true,
+  reactStrictMode: false,
```
Comment (3lLobo) [inline next.config.js]: Holdup, this is not legit. Why do we need this? 🤔

**Review by 3lLobo — CHANGES_REQUESTED** [review]
The Filetransfer part looks really good!
The changes in `contactSlice` and `Sidebar` are not necessary if you merge `main`.
Why do you want to change react-strict mode?
```bash
git switch main
git pull
git switch Filetransfer
git merge main
git push origin Filetransfer
```

File: app/contactSlice.js (hunk 1 above)
Comment (perassijulian) [reply app/contactSlice.js]: The changes that are not on filetransfer are like that cause stayed as they were before your changes

File: components/FileTransfer/index.js (hunk 2 above)
Comment (perassijulian) [reply components/FileTransfer/index.js:245]: I didn't know what was isOneHop. I've just copied an AddressTag

File: next.config.js (hunk 4 above)
Comment (perassijulian) [reply next.config.js]: If not when we go to FileTransfer tab getFiles is fired 2 times. As I've read when reactStrictMode: false the behaviour is the same as when in production and the main change is that useEffect doesn't trigger 2 times on the init

File: components/SideBar/index.js (hunk 3 above)
Comment (perassijulian) [reply components/SideBar/index.js]: I'm still super newbie with branches and merges. My bad

File: next.config.js (hunk 4 above)
Comment (3lLobo) [reply next.config.js]: Okay, I think there should be an easier way.
AFAIK strict mode is needed for debugging. Check if you can find another solution and leave this as plan b

**Pull request merged**